### PR TITLE
Preserve lights' color & temperature (#75)

### PR DESCRIPTION
**Pull request by ValShaped**
Modified update_entity to preserve color modes:
[color_temp, rgbww, rgbw, rgb, hs, xy]

### CODE REVIEW

**Review by slashback100 — APPROVED** [review]

File: custom_components/presence_simulation/__init__.py [hunk 1]
```diff
@@ -325,8 +325,28 @@ async def update_entity(entity_id, state):
...
-                service_data["rgb_color"] = state.attributes["rgb_color"]
+            # Preserve accurate color information, where applicable
+            # color modes "onoff", "white", etc. do not need preservation.
+            # see https://developers.home-assistant.io/docs/core/entity/light/#color-modes
+            # see https://developers.home-assistant.io/docs/core/entity/light/#turn-on-light-device
+            if "color_mode" in state.attributes:
+                match state.attributes["color_mode"]:
+                    # color temperature mode (mireds, kelvin)
+                    case "color_temp":
+                        service_data["color_temp"]  = state.attributes["color_temp"]
```
[the remark this answers is not in the document]
Comment (slashback100) [reply custom_components/presence_simulation/__init__.py:336]: Something like this should work
```
            if "color_mode" in state.attributes:
                _LOGGER.debug("Got attribute color_mode: %s", state.attributes["color_mode"])
                color_mode = state.attributes["color_mode"]
                if color_mode != "color_temp":
                    #attribute color_mode will be hs, rgb...
                    color_mode = color_mode+"_color"
                if color_mode in state.attributes:
                    service_data[color_mode] = state.attributes[color_mode]
```

**Review by slashback100 — CHANGES_REQUESTED** [review]

File: custom_components/presence_simulation/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ValShaped) [reply custom_components/presence_simulation/__init__.py:336]: The Rust developer in me is cringing, and that'd break for lights that expose color modes not explicitly handled in my initial PR (i.e. "white" and "onoff" which have no associated color), but yeah, that could work for RGB/WW, HS, and XY lights. I'll try it out on my instance.

[the remark this answers is not in the document]
Comment (ValShaped) [reply custom_components/presence_simulation/__init__.py]: Fixed in https://github.com/slashback100/presence_simulation/pull/76/commits/96a274bcd56e6ee04ce339927da56c92a9e55865

**Review by slashback100 — APPROVED** [review]

**Comment by slashback100** [comment]
Thanks !

**Pull request merged**